RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTExtraProducer.cc

```diff
@@ -0,0 +1,459 @@
+    for (const auto& tokenLabel : tokenLabels) {
+      auto handle = event.getHandle(tokenLabel.first);
+      auto recHits = filterRecHits(egTrigObjs, handle, *caloGeomHandle);
+      event.put(std::move(recHits), tokenLabel.second);
+    }
+  };
+  auto filterAndStore = [&event, this](const auto& egTrigObjs, const auto& tokenLabels, auto filterFunc) {
+    for (const auto& tokenLabel : tokenLabels) {
+      auto handle = event.getHandle(tokenLabel.first);
+      auto filtered = (this->*filterFunc)(egTrigObjs, handle, 0.4 * 0.4);
+      event.put(std::move(filtered), tokenLabel.second);
+    }
+  };
+
+  filterAndStoreRecHits(egTrigObjColls, tokens_.ecal);
+  filterAndStoreRecHits(egTrigObjColls, tokens_.hcal);
+  filterAndStore(egTrigObjColls, tokens_.pfClusIso, &EgammaHLTExtraProducer::filterPFClusIso);
+  filterAndStore(egTrigObjColls, tokens_.trks, &EgammaHLTExtraProducer::filterTrks);
+
+  for (size_t collNr = 0; collNr < egTrigObjColls.size(); collNr++) {
+    event.put(std::move(egTrigObjColls[collNr]), tokens_.egCands[collNr].second);
+  }
+}
+
+void EgammaHLTExtraProducer::setVars(
+    reco::EgTrigSumObj& egTrigObj,
+    const reco::RecoEcalCandidateRef& ecalCandRef,
+    const std::vector<edm::Handle<reco::RecoEcalCandidateIsolationMap>>& valueMapHandles) {
+  for (auto& valueMapHandle : valueMapHandles) {
+    auto mapIt = valueMapHandle->find(ecalCandRef);
+    if (mapIt != valueMapHandle->end()) {
+      std::string name = valueMapHandle.provenance()->moduleLabel();
+      if (!valueMapHandle.provenance()->productInstanceName().empty()) {
+        name += "_" + valueMapHandle.provenance()->productInstanceName();
+      }
+      egTrigObj.setVar(std::move(name), mapIt->val);
+    }
+  }
+}
+
+reco::GsfTrackRefVector EgammaHLTExtraProducer::matchingGsfTrks(
+    const reco::SuperClusterRef& scRef, const edm::Handle<reco::GsfTrackCollection>& gsfTrksHandle) {
+  if (!gsfTrksHandle.isValid()) {
+    return reco::GsfTrackRefVector();
+  }
+
+  reco::GsfTrackRefVector gsfTrkRefs(gsfTrksHandle.id());
+  for (size_t trkNr = 0; gsfTrksHandle.isValid() && trkNr < gsfTrksHandle->size(); trkNr++) {
+    reco::GsfTrackRef trkRef(gsfTrksHandle, trkNr);
+    edm::RefToBase<TrajectorySeed> seed = trkRef->extra()->seedRef();
+    reco::ElectronSeedRef eleSeed = seed.castTo<reco::ElectronSeedRef>();
+    edm::RefToBase<reco::CaloCluster> caloCluster = eleSeed->caloCluster();
+    reco::SuperClusterRef scRefFromTrk = caloCluster.castTo<reco::SuperClusterRef>();
+    if (scRefFromTrk == scRef) {
+      gsfTrkRefs.push_back(trkRef);
+    }
+  }
+  return gsfTrkRefs;
+}
+
+void EgammaHLTExtraProducer::setGsfTracks(reco::EgTrigSumObj& egTrigObj,
+                                          const edm::Handle<reco::GsfTrackCollection>& gsfTrksHandle) {
+  reco::GsfTrackRefVector gsfTrkRefs = matchingGsfTrks(egTrigObj.superCluster(), gsfTrksHandle);
+  egTrigObj.setGsfTracks(std::move(gsfTrkRefs));
```

Comment:
How about
```suggestion
  egTrigObj.setGsfTracks(matchingGsfTrks(egTrigObj.superCluster(), gsfTrksHandle));
```
?

Reply:
yes thats better I'm not sure what I was thinking, thanks!